src/libsync/account.cpp

```diff
@@ -201,6 +203,30 @@ void Account::setCredentials(AbstractCredentials *cred)
         this, &Account::slotCredentialsFetched);
     connect(_credentials.data(), &AbstractCredentials::asked,
         this, &Account::slotCredentialsAsked);
+
+    trySetupPushNotifications();
+}
+
+void Account::trySetupPushNotifications()
+{
+    if (_capabilities.availablePushNotifications() & PushNotificationType::Files) {
+        if (!_pushNotifications) {
+            _pushNotifications = new PushNotifications(this, this);
+            connect(_pushNotifications, &PushNotifications::ready, this, [this]() { emit pushNotificationsReady(this); });
+            connect(_pushNotifications, &PushNotifications::connectionLost, this, [this]() {
+                qCInfo(lcAccount) << "Delete push notifications object because connection lost";
+                _pushNotifications->deleteLater();
+                _pushNotifications = nullptr;
+            });
+            connect(_pushNotifications, &PushNotifications::authenticationFailed, this, [this]() {
```

Comment:
I'd reuse the same lambda in that connect and the connect just before. No need to duplicate that code just for a single log line difference (especially since the PushNotifications object logs before emitting those signals anyway).